src/KeywordCalls.jl

```diff
@@ -182,8 +182,8 @@ function _kwalias(__module__, fsym, aliasmap)
 end
 
 # See https://github.com/cscherrer/KeywordCalls.jl/issues/22
-@inline instance_type(f::F) where {F<:Function} = F
-@inline instance_type(f::UnionAll) = Type{f}
-
+@inline instance_type(::F) where {F} = F
+@inline instance_type(::UnionAll) = Type{<:T}
```

Comment:
Why the `<:T`?

Reply:
I realized we were missing cases like
```julia
julia> typeof(rand(2)) isa Type{Array{Float64}}
false

julia> typeof(rand(2)) isa Type{<:Array{Float64}}
true
```

Reply:
Why would that be expected to hold?

Reply:
I'm not sure. `Core.Typeof` passes the tests, so let's try that and come back to this if we need to